pkg/tfbridge/schema_test.go

```diff
@@ -411,6 +411,135 @@ func TestMakeTerraformInputMixedMaxItemsOne(t *testing.T) {
 	}
 }
 
+func TestMaxItemsOneEmptyOldState(t *testing.T) {
```

Comment:
Can we name this test in a way that tells the reader what it's testing, perhaps assisted by a comment? Something like this:

Test that makeTerraformInputs, makeTerraformInputsWithDefaults work well with MaxItems=1 properties.

- missing MaxItems=1 properties should present to TF as empty collections when applying the actual changes (makeTerraformInputsWithDefaults)
- missing MaxItems=1 properties should present to TF as missing when running validators (makeTerraformInputs)

Nit: it could also benefit from a table-driven test pattern to eliminate a little bit of code repetition.